osbrain/agent.py

```diff
@@ -204,6 +214,19 @@ class Agent():
         When set to `True`, the agent will continue executing the main loop.
     running : bool
         Set to `True` if the agent is running (executing the main loop).
+    _async_req_uuid : dict
+        Stores a unique identifier for every REQ or PULL connection.
+    _async_req_handler : dict
+        Stores the handler for every active REQ or PULL connection.
+    _kill_now : bool
+        Used internally to synchronously shut down the agent.
+    _DEBUG : bool
+        Whether to print debug level messages.
+    _pending_requests : dict
+        Stores the handler for every REQ message sent while waiting for the
+        reply.
```

Comment:
I'd say "Stores pending (without reply) asynchronous requests. The asynchronous request UUID is used as key and the handler is set as value." Note that "asynchronous requests" are not the same as REQ messages. Need to add documentation on channels, which is currently missing... :sweat:

Reply:
That would definitely be helpful for some of us 👍 

There are still a couple of things that are not fully clear to me.

Reply:
Yeah, will try to do that today/tomorrow.